pkg/ring/lifecycler.go

```diff
@@ -147,7 +147,8 @@ func NewLifecycler(cfg LifecyclerConfig, flushTransferer FlushTransferer, ringNa
 	}
 	port := GetInstancePort(cfg.Port, cfg.ListenPort)
 	codec := GetCodec()
-	store, err := kv.NewClient(cfg.RingConfig.KVStore, codec)
+	// Suffix all client names with "-lifecycler" to denote this kv client is used by the lifecycler
+	store, err := kv.NewClient(ringName+"-lifecycler", cfg.RingConfig.KVStore, codec, nil)
```

Comment:
This is a great opportunity to pass registrerer to lifecycler and move lifecycler metrics away from global variables.

Reply:
But maybe in another PR?

Reply:
I will pass the Registerer but keep the lifecycler metrics global for now.